hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/FileSystemUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.util;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.nio.file.WatchEvent;
+
+
+public final class FileSystemUtils {
+
+    @SuppressFBWarnings("MS_MUTABLE_ARRAY")
+    // Modifiers for file watch service to achieve the highest possible sensitivity.
+    // Background: Java 7 SE defines no standard modifiers for a watch service. However some JDKs use internal
+    // modifiers to increase sensitivity. This field contains modifiers to be used for highest possible sensitivity.
+    // It's JVM-specific and hence it's just a best-effort. The caller is not allowed to modify the array.
+    // I believe this is useful on platforms without native watch service (or where Java does not use them) e.g. MacOSX
+    public static final WatchEvent.Modifier[] HIGH_SENSITIVITY_MODIFIERS = getHighSensitivityModifiersIfAvailable();
+
+    private static final String SENSITIVITY_MODIFIER_CLASSNAME = "com.sun.nio.file.SensitivityWatchEventModifier";
+
+    private FileSystemUtils() {
+
+    }
+
+    private static WatchEvent.Modifier[] getHighSensitivityModifiersIfAvailable() {
+        WatchEvent.Modifier modifier = ReflectionUtils.readStaticFieldOrNull(SENSITIVITY_MODIFIER_CLASSNAME, "HIGH");
+        if (modifier == null) {
+            //bad luck, we did not find the modifier
+            return new WatchEvent.Modifier[0];
+        }
+        WatchEvent.Modifier[] modifiers = new WatchEvent.Modifier[1];
+        modifiers[0] = modifier;
+        return modifiers;
```

Comment:
We can do just `return new WatchEvent.Modifier[] {modifier}`.
Also, I would inline the whole `FileSystemUtils` class into `StreamFilesP`, it's too specific for this processor.

Reply:
done